src/connectedk8s/azext_connectedk8s/_troubleshootutils.py

```diff
@@ -0,0 +1,567 @@
+                    pod_name = each_pod.metadata.name
+
+                    if(pod_name.startswith(job_name)):
+
+                        # Creating a text file with the name of the container and adding that containers logs in it
+                        diagnoser_container_log = corev1_api_instance.read_namespaced_pod_log(name=pod_name, container="networktest1", namespace=namespace)
+
+                # Clearing all the resources after fetching the diagnoser container logs
+                subprocess.run(command, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
+
+            except Exception as e:
+                subprocess.run(command, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
+
+    except KeyboardInterrupt:
+        # If process terminated by user then delete the resources if any added to the cluster.
+        subprocess.run(command, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
+
+    except Exception as e:
+        logger.warning("An expection has occured while trying to execute the diagnoser job in the cluster. Exception: {}".format(str(e)) + "\n")
+        subprocess.run(command, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
+
+    return diagnoser_container_log
+
+
+def diagnoser_container_check(corev1_api_instance, batchv1_api_instance, filepath_with_timestamp , storage_space_available, namespace, absolute_path):
+
+    try:
+        # Setting DNS and Outbound Check as working
+        dns_check = True
+        outbound_connectivity_check = True
+
+        # Executing the Diagnoser job and fetching the logs obtained
+        diagnoser_container_log = executing_diagnoser_job(corev1_api_instance, batchv1_api_instance, namespace, absolute_path)
+
+        if(diagnoser_container_log != ""):
+            dns_check, storage_space_available = check_cluster_DNS(diagnoser_container_log, filepath_with_timestamp, storage_space_available)
+            outbound_connectivity_check, storage_space_available = check_cluster_outbound_connectivity(diagnoser_container_log, filepath_with_timestamp, storage_space_available)
+
+        if(dns_check and outbound_connectivity_check):
+            return True, storage_space_available
+        return False, storage_space_available
+
+    except Exception as e:
+        logger.warning("An expection has occured while trying to perform diagnoser continer check on the cluster. Exception: {}".format(str(e)) + "\n")
+
+    return True, storage_space_available
+
+
+def check_cluster_DNS(diagnoser_container_log, filepath_with_timestamp, storage_space_available):
+
+    try:
+        # To retreive only the DNS lookup result from the diagnoser container logs
+        dns_check = diagnoser_container_log[0:len(diagnoser_container_log) - 5:]
+
+        # Validating if DNS is working or not and displaying proper result
+        if("NXDOMAIN" in dns_check or "connection timed out" in dns_check):
```

Comment:
what hostnames do you try to resolve for DNS check? i think we should try the resolution for arc network requirement urls. and point then to the network requirement page if this check fails

Reply:
Will raise the PR for the script and dockerfile today itself so that you get more clear idea.